merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/models/HasuraAction.java

```diff
@@ -13,12 +13,14 @@ public sealed interface Input { }
 
   public record MissionModelInput(String missionModelId) implements Input { }
   public record PlanInput(PlanId planId) implements Input { }
+  public record ConstraintViolationsInput(PlanId planId, Optional<SimulationDatasetId> simulationDatasetId) implements Input { }
   public record ActivityInput(String missionModelId,
                               String activityTypeName,
                               Map<String, SerializedValue> arguments) implements Input {}
   public record MissionModelArgumentsInput(String missionModelId,
                               Map<String, SerializedValue> arguments) implements Input {}
   public record UploadExternalDatasetInput(PlanId planId,
+                                           Optional<SimulationDatasetId> associatedSimulationDatasetId,
```

Comment:
Based on the fact that if `associatedSimulationDatasetId` is defined, a disjoint set of logic appears to be run, I wonder if it would make more sense to create a new action with new functions (ie `getAllPlanDatasetsForSimulation`) rather than add an Optional parameter to our existing actions.

Reply:
Related to this, in the Java methods, if the Optional argument is defined, then the method `getAllPlanDatasetsForPlan` does not do what its name implies it does.

Reply:
I agree and this was really my main concern with proposed implementation. My reasoning was that a relatively niche feature would be better suited as an optional parameter for an existing API endpoint vs defining an entirely new endpoint for both `uploadExternalDataset` and `constraintViolations`

Reply:
Additionally, I modeled this after `ConstraintsInput`:

https://github.com/NASA-AMMOS/aerie/blob/8cfdc83f0dbc0d61621d489f068e529dc85fba50/merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/models/HasuraAction.java#L27

where an optional is used to augment an API endpoint instead of creating two endpoints.

Reply:
Perhaps then just splitting the `getAllPlanDatasetsForSimulation` part of `getAllPlanDatasetsForPlan` into its method would be sufficient then, as it would resolve the main issue of `getAllPlanDatasetsForPlan` being an inaccurate method name.

Reply:
change has been made!